### ISSUE TITLE
Regression: Star Wars: The Force Unleashed II Demo [NPEB90251] (#6914)

### ISSUE THREAD

**Issue by Kravickas** [RPCS3/rpcs3#6971]
Star Wars: The Force Unleashed II Demo [NPEB90251] regressed after #6914 - Vulkan only

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3827495/RPCS3.log.gz)

> F {RSX [0x00059e4]} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
> (in file C:\projects\rpcs3\rpcs3\Emu\RSX\VK\VKHelpers.h:1144)

as soon as it should display splash screens

**Comment by kd-11** [comment]
Need a log with debug output enabled (requires vulkan SDK)

**Comment by Kravickas** [comment]
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3827981/RPCS3.log.gz)

**Comment by Senninz8** [comment]
Can confirm regression. Same problem with NINJA GAIDEN Σ2 [NPUB30852] during in-game cutscenes.

**Comment by legend800** [comment]
Try with this:
https://github.com/RPCS3/rpcs3/pull/6862

**Comment by pitkevich80** [comment]
> Can confirm regression. Same problem with NINJA GAIDEN Σ2 [NPUB30852] during in-game cutscenes.

Exactly the same problem with Ninja Gaiden Sigma 2. 
Also freeze with RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST) 
ON cut scenes only. (( 
OPenGL work stable, but slow ((

**Comment by kd-11** [comment]
Yea, these games are buggy, so you see that annoying error about "invalid 2D region". Simple enough to fix, a patch will be made available soon.

**Comment by jeremybris** [comment]
This also effects Beyond Good & Evil HD [NPUB30394].

**Comment by kd-11** [comment]
The fix has been available for some time, just cleaning up for proper PR in the next 24h.

**Comment by legend800** [comment]
Still not fixed on your latest build if this is the correct build: 
https://ci.appveyor.com/project/kd-11/rpcs3

F {RSX [0x00059e4]} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file C:\projects\rpcs3\rpcs3\Emu\RSX\VK\VKHelpers.h:1178)

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/3853844/RPCS3.zip)

**Comment by pitkevich80** [comment]
Same error with with Ninja Gaiden Sigma 2 too.. 
(VK_ERROR_DEVICE_LOST)

**Comment by kd-11** [comment]
Not the correct build.

**Comment by pitkevich80** [comment]
So next build?

**Comment by legend800** [comment]
Sorry forgot to mention same thing with your Optimizations branch. Figured it wasn't in your MSAA branch. Maybe it's in a top secret branch then... :)

**Comment by pitkevich80** [comment]
YAY! It finally work now ;) Almost perfectly ..

**Comment by Shoegzer** [comment]
@kd-11 reporting from #6983 since it was a duplicate of this issue.  Commit cd6b6c8a4f19595ac339efbf86602ecf59b5a497 fixed the issue with Far Cry hanging on initial startup, but it still hangs at the loading screen.  I've seen it do this before, but never this consistently (I tried 5-6 times and it hung every time). It might be sourced from the same problem as #5968 and if so, it's probably entirely unrelated so please disregard.  Thanks for any help here.

Log of my most recent attempt: [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3866332/RPCS3.log.gz)

Tested on Linux Mint 19.2 Cinnamon x64 with kernel 5.0, NVIDIA GTX 1070M driver 440.26
